src/core/sock/sock-app.cpp

```diff
@@ -253,7 +254,7 @@ static int init_worker(int worker_id, int listen_fd)
         if (child_sock_fd_api) {
             child_sock_fd_api->copy_sockopt_fork(parent_sock_fd_api);
 
-            ret = bind(listen_fd, sa.get_p_sa(), sa_len);
+            ret = bind_internal(child_sock_fd_api, sa.get_p_sa(), sa_len);
```

Comment:
Shouldn't it be listen_fd?

Reply:
No. Because of lines 253-254, the listen_fd and child_socket_fd_api reference the same object. The first reference is with a file descriptor, and the other is with a pointer to the object memory address.